src/bin/wallet.rs

```diff
@@ -245,7 +243,7 @@ fn process_command(
         }
         WalletCommand::Balance => {
             println!("Balance requested...");
-            let balance = client.poll_get_balance(&config.id.pubkey());
+            let balance = client.poll_balance_with_timeout(&config.id.pubkey(), 100, 1000);
```

Comment:
The change I proposed was backward compatible, hence the name change to the version needing the new parameter.